src/style/colors.ts

```diff
@@ -220,38 +220,19 @@ export class Colors {
     return this.shouldSupportDarkMode && Scheme.getSchemeType() === 'dark' ? _.reverse(palette) : palette;
   });
 
-  generateDesignTokens(color: string, mode: 'light' | 'dark' = 'light') {
-    return mode === 'light' ? this.generateLightModeTokens(color) : this.generateDarkModeTokens(color);
-  }
-
-  private generateLightModeTokens(color: string) {
-    const colorPalette: string[] = this.generatePalette(color);
-    const color30 = colorPalette[2];
-    const color50 = colorPalette[4];
-    const color70 = colorPalette[6];
-    const color80 = colorPalette[7];
-
-    const mainColor = this.isDark(color) ? color : color30;
-
-    return {
-      $backgroundPrimaryHeavy: mainColor,
-      $backgroundPrimaryLight: color80,
-      $backgroundPrimaryMedium: color70,
-      $iconPrimary: mainColor,
-      $iconPrimaryLight: color50,
-      $textPrimary: mainColor,
-      $outlinePrimary: mainColor
-    };
-  }
-
-  private generateDarkModeTokens(color: string) {
-    const colorPalette: string[] = _.reverse(this.generatePalette(color));
+  private generateDesignTokens({primaryColor, dark}: {primaryColor: string; dark?: boolean}) {
+    const colorPalette: string[] = dark
+      ? _.reverse(this.generatePalette(primaryColor))
+      : this.generatePalette(primaryColor);
     const color30 = colorPalette[2];
     const color50 = colorPalette[4];
     const color70 = colorPalette[6];
     const color80 = colorPalette[7];
 
-    const mainColor = this.isDark(color) ? color30 : color;
+    let mainColor = this.isDark(primaryColor) ? primaryColor : color30;
```

Comment:
I rewrote this condition and ended up with this one:
```
const mainColor = this.isDark(primaryColor) === !!dark ? color30 : primaryColor
```
because I want to use color30 only if the color is light in light mode or dark in dark mode, but I'm not sure if it makes the code more readable than that:

```
let mainColor = isPrimaryColorDark ? primaryColor : color30;
if (dark) {
  mainColor = isPrimaryColorDark ? color30 : primaryColor;
}
```
WDYT?

Reply:
if `cont isPrimaryColorDark = this.isDark(primaryColor)` is the line before then it readable to do:
`let mainColor = isPrimaryColorDark ? primaryColor : color30;
if (dark) {
  mainColor = isPrimaryColorDark ? color30 : primaryColor;
}`